### PR TITLE
remove lock from url

### LINKED ISSUE #511: API design problem causes lock copy
Currently,  dubbo-go NewURL returns a non-ptr instance of URL，NewURLWithOptions returns a ptr of URL, 

```go
func NewURL(urlString string, opts ...option) (URL, error) {

func NewURLWithOptions(opts ...option) *URL {
```

and Register API uses a non-ptr instance of URL:

```go
func (r *BaseRegistry) Register(conf common.URL) error {
```

This will cause a lock copy, because URL contains a mutex field:

```go
type baseUrl struct {
	Protocol string
	Location string // ip+port
	Ip       string
	Port     string

	paramsLock   sync.RWMutex   <------ here
	params       url.Values
	PrimitiveURL string
}
```

The Register API should accept a pointer to avoid this

### PR DESCRIPTION
**Pull request by flycash**
**What this PR does**:

Remove the RWLock from baseURL

**Which issue(s) this PR fixes**:

As my understanding, the URL struct should be immutable. But there are some Set* and Add* methods which can modify the URL's instance.

However, we need those methods to create a URL's instance. But the instance should not be modified anymore after finish initialization.

Another concern is that when we try to clone a URL instance, the lock will be cloned too. It will cause some concurrent problems.

Thus remove this lock will be more reasonable. 

As the expense of that, the developer should be more careful to use the URL, especially when they decide to invoke Set* or Add* method.

Fixes #511 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: common/url.go [hunk 1]
```diff
@@ -425,8 +424,8 @@ func (c URL) GetParams() url.Values {
 
 // GetParamAndDecoded ...
 func (c URL) GetParamAndDecoded(key string) (string, error) {
-	c.paramsLock.RLock()
-	defer c.paramsLock.RUnlock()
+	// c.paramsLock.RLock()
+	// defer c.paramsLock.RUnlock()
```
Comment (zouyx) [inline common/url.go:428]: should you delete it?

File: common/url.go [hunk 2]
```diff
@@ -409,8 +408,8 @@ func (c *URL) RangeParams(f func(key, value string) bool) {
 
 // GetParam ...
 func (c URL) GetParam(s string, d string) string {
-	c.paramsLock.RLock()
-	defer c.paramsLock.RUnlock()
+	// c.paramsLock.RLock()
+	// defer c.paramsLock.RUnlock()
```
Comment (zouyx) [inline common/url.go:412]: should you delete it?

**Comment by zouyx** [comment]
resolve conflicts

File: common/url.go (hunk 2 above)
Comment (flycash) [reply common/url.go]: done

File: common/url.go (hunk 1 above)
Comment (flycash) [reply common/url.go]: done

**Pull request merged**